private/tools/java/com/github/bazelbuild/rules_jvm_external/javadoc/JavadocJarMaker.java

```diff
@@ -175,7 +185,27 @@ public static void main(String[] args) throws IOException {
 
       options.addAll(Arrays.asList("-d", outputTo.toAbsolutePath().toString()));
 
-      sources.forEach(obj -> options.add(obj.getName()));
+      // sourcepath and subpackages should work in most cases.
+      // A known edge case is when the package names don't match the directory structure.
+      // For example `OneDep.java` in `tests/integration/maven_bom/OneDep.java` has a package
+      // of "com.github.bazelbuild.rules_jvm_external.example.maven_bom" but the file is in
+      // `tests/integration/maven_bom/OneDep.java`. I added the condition here to handle that case
+      // for now.
+      if (!excludedPackages.isEmpty()) {
```

Comment:
you'll have to validate this but I think In java the rule is:
* source files don't have to match package directory
* class files DO have to match package directory